ui/src/routes/admin/plugins/index.tsx

```diff
@@ -41,24 +50,92 @@ const PluginListCard = ({
   title,
   description,
   plugins,
-}: PluginListCardProps) => (
-  <Card className='mb-4 h-fit'>
-    <CardHeader>
-      <CardTitle>{title}</CardTitle>
-      <CardDescription>{description}</CardDescription>
-    </CardHeader>
-    <CardContent>
-      {plugins.map((plugin) => (
-        <Card key={plugin.id} className='mb-2'>
-          <CardHeader>
-            <CardTitle>{plugin.displayName}</CardTitle>
-            <CardDescription>{plugin.description}</CardDescription>
-          </CardHeader>
-        </Card>
-      ))}
-    </CardContent>
-  </Card>
-);
+}: PluginListCardProps) => {
+  const enablePlugin =
+    usePluginsServicePostApiV1AdminPluginsByPluginTypeByPluginIdEnable();
+  const disablePlugin =
+    usePluginsServicePostApiV1AdminPluginsByPluginTypeByPluginIdDisable();
+
+  const handleToggle = (plugin: PluginDescriptor) => {
+    if (plugin.enabled) {
+      disablePlugin.mutate(
+        { pluginType: plugin.type, pluginId: plugin.id },
+        {
+          onSuccess: () => {
+            toast.info('Disable Plugin', {
+              description: `Plugin "${plugin.displayName}" disabled successfully.`,
+            });
+            queryClient.invalidateQueries({
+              queryKey: [usePluginsServiceGetApiV1AdminPluginsKey],
+            });
+          },
+          onError(error: unknown) {
+            const apiError = error as ApiError;
```

Comment:
@Etsija This was the one tiny bit I didn't like about the UI code, everywhere else it's `onError(error: ApiError)` but I couldn't figure out how to make this work here, hence the cast. So if you have an idea how to improve it I would be interested.